docker/Dockerfile

```diff
@@ -6,61 +6,77 @@ MAINTAINER Hector Fernandez <hfernand@redhat.com>
-RUN java -jar /usr/share/java/saxon.jar -s:/opt/jboss/keycloak/standalone/configuration/standalone.xml -xsl:/opt/jboss/keycloak/changeDatabase.xsl -o:/opt/jboss/keycloak/standalone/configuration/standalone.xml && \
-    java -jar /usr/share/java/saxon.jar -s:/opt/jboss/keycloak/standalone/configuration/standalone-ha.xml -xsl:/opt/jboss/keycloak/changeDatabase.xsl -o:/opt/jboss/keycloak/standalone/configuration/standalone-ha.xml && \
-    rm /opt/jboss/keycloak/changeDatabase.xsl
+ADD changeDatabase.xsl $KEYCLOAK_HOME
+RUN java -jar /usr/share/java/saxon.jar -s:$KEYCLOAK_HOME/standalone/configuration/standalone.xml -xsl:$KEYCLOAK_HOME/changeDatabase.xsl -o:$KEYCLOAK_HOME/standalone/configuration/standalone.xml && \
+    java -jar /usr/share/java/saxon.jar -s:$KEYCLOAK_HOME/standalone/configuration/standalone-ha.xml -xsl:$KEYCLOAK_HOME/changeDatabase.xsl -o:$KEYCLOAK_HOME/standalone/configuration/standalone-ha.xml && \
+    rm $KEYCLOAK_HOME/changeDatabase.xsl
 
 ENV PSQL_JDBC_VERSION 9.3-1104-jdbc41
 
-RUN mkdir -p $JBOSS_HOME/modules/system/layers/base/org/postgresql/jdbc/main; cd $JBOSS_HOME/modules/system/layers/base/org/postgresql/jdbc/main; curl -O http://central.maven.org/maven2/org/postgresql/postgresql/$PSQL_JDBC_VERSION/postgresql-$PSQL_JDBC_VERSION.jar
-ADD module-postgres-jdbc.xml $JBOSS_HOME/modules/system/layers/base/org/postgresql/jdbc/main/
-RUN envsubst < $JBOSS_HOME/modules/system/layers/base/org/postgresql/jdbc/main/module-postgres-jdbc.xml > $JBOSS_HOME/modules/system/layers/base/org/postgresql/jdbc/main/module.xml
-RUN rm $JBOSS_HOME/modules/system/layers/base/org/postgresql/jdbc/main/module-postgres-jdbc.xml
-RUN sed -ie 's@\(</dependencies>\)@    <module name="org.postgresql.jdbc"/>\n    \1@' /opt/jboss/keycloak/modules/system/layers/base/org/jgroups/main/module.xml
+RUN mkdir -p $KEYCLOAK_HOME/modules/system/layers/base/org/postgresql/jdbc/main
+ADD module-postgres-jdbc.xml $KEYCLOAK_HOME/modules/system/layers/base/org/postgresql/jdbc/main/
+RUN cd $KEYCLOAK_HOME/modules/system/layers/base/org/postgresql/jdbc/main && \
+  curl -O http://central.maven.org/maven2/org/postgresql/postgresql/$PSQL_JDBC_VERSION/postgresql-$PSQL_JDBC_VERSION.jar && \
+  envsubst < $KEYCLOAK_HOME/modules/system/layers/base/org/postgresql/jdbc/main/module-postgres-jdbc.xml > $KEYCLOAK_HOME/modules/system/layers/base/org/postgresql/jdbc/main/module.xml && \
+  rm $KEYCLOAK_HOME/modules/system/layers/base/org/postgresql/jdbc/main/module-postgres-jdbc.xml
+
+ENV JGROUPS_K8S_VERSION 0.9.2
+ENV OAUTH_VERSION 20090531
+
+RUN mkdir -p $KEYCLOAK_HOME/modules/system/layers/base/org/jgroups/kubernetes/kubernetes/main
+ADD module-jgroups-k8s.xml $KEYCLOAK_HOME/modules/system/layers/base/org/jgroups/kubernetes/kubernetes/main
+RUN cd $KEYCLOAK_HOME/modules/system/layers/base/org/jgroups/kubernetes/kubernetes/main && \
+ curl -O http://central.maven.org/maven2/org/jgroups/kubernetes/kubernetes/$JGROUPS_K8S_VERSION/kubernetes-$JGROUPS_K8S_VERSION.jar && \
```

Comment:
We don't have those deps as part of the KC poms. We would have to change that. Not sure how much would that improve. CURL makes it simple here.